datashape/discovery.py

```diff
@@ -160,13 +172,10 @@ def discover(s):
 
     try:
         d = dateparse(s)
-        if not d.time():
-            return date_
-        if not d.date():
-            return time_
-        return datetime_
-    except:
+    except ValueError:
         pass
+    else:
+        return date_ if not d.time() else datetime_
```

Comment:
how does that ever happen? how do you construct a `datetime.date` object that is falsey?